File: pgm/ABC_Dart/ABC_309_A.dart
// import 'dart:collection';
import 'dart:io';
// import 'dart:math';
import 'dart:typed_data';

void main() {
  final [int A, int B] = stdin
      .readLineSync()!
      .split(' ')
      .map(int.parse)
      .toList();
  const String Yes = 'Yes';
  const String No = 'No';
  const int one = 1;
  const int three = 3;
  const int zero = 0;
  const int H = three;
  const int W = three;
  final List<Int32List> N = List.generate(
    H,
    (y) => Int32List.fromList(List.generate(W, (x) => y * W + x + one)),
  );
  // 右、左
  final Int32List dy = Int32List.fromList([zero, zero]);
  final Int32List dx = Int32List.fromList([one, -one]);
  String adjacent = No;
  loop:
  for (int y = zero; y < H; y++) {
    for (int x = zero; x < W; x++) {
      if (N[y][x] == A) {
        for (int k = zero; k < dy.length; k++) {
          final int ny = y + dy[k];
          final int nx = x + dx[k];
          if (ny >= zero && ny < H && nx >= zero && nx < W) {
            if (N[ny][nx] == B) {
              adjacent = Yes;
              break loop;
            }
          }
        }
        break loop;
      }
    }
  }
  print(adjacent);
  exit(0);
}

File: pgm/ABC_Kotlin/ABC_309_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.StringTokenizer

// import kotlin.math.*

const val Yes: String = "Yes"
const val No: String = "No"
const val space: String = " "
const val one: Int = 1
const val three: Int = 3
const val zero: Int = 0
const val H:Int = three
const val W:Int = three

fun main() {
    val (A: Int, B: Int) = readln().split(space).map { it.toInt() }
    val N: Array<IntArray> = Array(H) { y ->
        IntArray(W) { x -> 
            y * W + x + one 
        }
    }
    // 右、左
    val dy:IntArray = intArrayOf(zero, zero)
    val dx:IntArray = intArrayOf(one, -one)
    var adjacent:String = No
    loop@for (y in N.indices) {
        for (x in N[y].indices) {
            if (N[y][x] == A) {
                for (k in dy.indices) {
                    val ny = y + dy[k]
                    val nx = x + dx[k]
                    if (ny in N.indices && nx in N[y].indices) {
                        if (N[ny][nx] == B) {
                            adjacent = Yes
                            break@loop
                        }
                    }
                }
                break@loop
            }
        }
    }
    println(adjacent)
    kotlin.system.exitProcess(0)
}